### ISSUE TITLE
AppinioSocialShare().shareToTwitter

### ISSUE THREAD

**Issue by faturah9** [appinioGmbH/flutter_packages#95]
I tried AppinioSocialShare().shareToTwitter

in the emulator
and in the emulator, there is no twitter application installed, why does it error?

<img width="1392" alt="image" src="https://user-images.githubusercontent.com/31773595/236377496-30f99a70-0bdb-487b-8c35-fa6595c4055c.png">

**Comment by khanmujeeb687** [comment]
Hey @faturah9 , you are supposed to have the twitter app installed to share a file to twitter.  otherwise, this error will occur.

**Comment by faturah9** [comment]
> Hey @faturah9 , you are supposed to have the twitter app installed to share a file to twitter. otherwise, this error will occur.

thanks for your response, 
but What if the user doesn't have Twitter installed on their device? Will it be a force close?

**Comment by khanmujeeb687** [comment]
Yes, the app may force close in this case. To avoid such situations we have `getInstalledApps ` method to check which apps are installed. You can use this feature to prevent calling apps that are not installed.  Otherwise, checking every time if the app is installed can increase the time to open the other app.

**Comment by khanmujeeb687** [comment]
> > Hey @faturah9 , you are supposed to have the twitter app installed to share a file to twitter. otherwise, this error will occur.
> 
> thanks for your response, but What if the user doesn't have Twitter installed on their device? Will it be a force close?

Yes, the app may force close in this case. To avoid such situations we have `getInstalledApps ` method to check which apps are installed. You can use this feature to prevent calling apps that are not installed.  Otherwise, checking every time if the app is installed can increase the time to open the other app.

**Comment by khanmujeeb687** [comment]
> > Hey @faturah9 , you are supposed to have the twitter app installed to share a file to twitter. otherwise, this error will occur.
> 
> thanks for your response, but What if the user doesn't have Twitter installed on their device? Will it be a force close?

No this will not force close the app, Because we have try catch on every platform-specific implementation. And to avoid calling apps that are not installed we have getInstalledApps  method to check which apps are installed. You can use this feature to prevent calling apps that are not installed.
